tests/appsec/iast/aspects/test_re_aspects.py

```diff
@@ -491,3 +492,50 @@ def test_re_finditer_aspect_not_tainted():
     assert isinstance(res_iterator, typing.Iterator)
     for i in res_iterator:
         assert not is_pyobject_tainted(i)
+
+
+def test_re_match_getitem_aspect_tainted_string_re_object():
+    tainted_isaac_newton = taint_pyobject(
+        pyobject="Isaac Newton, physicist",
+        source_name="test_re_match_groups_aspect_tainted_string",
+        source_value="Isaac Newton, physicist",
+        source_origin=OriginType.PARAMETER,
+    )
+
+    re_obj = re.compile(r"(\w+) (\w+)")
+
+    re_match = re_match_aspect(None, 1, re_obj, tainted_isaac_newton)
+
+    isaac_newton = index_aspect(re_match, 0)
+    assert isaac_newton == "Isaac Newton"
+
+    isaac = index_aspect(re_match, 1)
+    assert isaac == "Isaac"
+
+    newton = index_aspect(re_match, 2)
+    assert newton == "Newton"
+
+    assert is_pyobject_tainted(isaac_newton)
+    assert is_pyobject_tainted(isaac)
+    assert is_pyobject_tainted(newton)
+
+
+def test_re_match_getitem_aspect_not_tainted_string_re_object():
+    not_tainted_isaac_newton = "Isaac Newton, physicist"
+
+    re_obj = re.compile(r"(\w+) (\w+)")
+
+    re_match = re_match_aspect(None, 1, re_obj, not_tainted_isaac_newton)
```

Comment:
It's ok a unit test but we need to test the AST replacement too. We need to add a function and patching it like: https://github.com/DataDog/dd-trace-py/blob/main/tests/appsec/iast/aspects/test_add_aspect_fixtures.py#L12